dingo/gw/inference/make_pp_plot.py

```diff
@@ -0,0 +1,173 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import stats
+import pandas as pd
+import argparse
+import textwrap
+
+
+def weighted_percentile_of_score(samples, value, weights=None):
+    """
+    Compute the percentile rank of a score relative to a list of scores
+    with attached weights.
+    
+    Parameters
+    ----------
+    samples: array-like
+        Array to which `value` is compared to
+    value: float 
+        A given score to compute the percentile for
+    weights: array-like; optional
+        Array of weights
+    
+    Returns
+    -------
+    percentile: float
+    """
+    if weights is None:
+        weights = np.ones_like(samples)
+    else:
+        if len(weights) != len(samples):
+            raise ValueError('weights and samples need to have the same length!')
+
+    # Sort the samples and weights
+    sorted_indices = np.argsort(samples)
+    sorted_samples = samples[sorted_indices]
+    sorted_weights = weights[sorted_indices]
+
+    # Calculate the cumulative sum of the weights and normalize to get the CDF
+    cdf = np.cumsum(sorted_weights) / sorted_weights.sum()
+
+    # Find the index of the first element in the sorted samples array
+    # that is greater than or equal to the given value
+    index = np.searchsorted(sorted_samples, value)
+
+    # Look up the corresponding CDF value at that index to get the percentile
+    percentile = cdf[index] * 100
```

Comment:
For `value` lying between two samples, this gives the percentile of the larger sample. There are other possibilities, e.g., given as options to the `method` argument of [numpy.percentile()](https://numpy.org/doc/stable/reference/generated/numpy.percentile.html). Are we doing the same as Bilby?